mapper-pojo/src/main/java/org/hibernate/search/v6poc/entity/pojo/model/path/impl/BoundPojoModelPathValueNode.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.v6poc.entity.pojo.model.path.impl;
+
+import org.hibernate.search.v6poc.entity.pojo.extractor.impl.BoundContainerValueExtractorPath;
+import org.hibernate.search.v6poc.entity.pojo.extractor.impl.ContainerValueExtractorPath;
+
+/**
+ * @param <T> The property holder type of this node, i.e. the type from which the property is retrieved.
+ * @param <P> The property type of this node, i.e. the type of the property from which the values are extracted.
+ * @param <V> The value type of this node, i.e. the type of values extracted from the property.
+ */
+public class BoundPojoModelPathValueNode<T, P, V> extends BoundPojoModelPath {
+
+	private final BoundPojoModelPathPropertyNode<T, P> parent;
+	private final BoundContainerValueExtractorPath<P, V> boundExtractorPath;
+	private BoundPojoModelPathTypeNode<V> elementTypePathNode;
+
+	BoundPojoModelPathValueNode(BoundPojoModelPathPropertyNode<T, P> parent,
+			BoundContainerValueExtractorPath<P, V> boundExtractorPath) {
+		this.parent = parent;
+		this.boundExtractorPath = boundExtractorPath;
+	}
+
+	/**
+	 * @return The model path to the property from which the value represented by this node is extracted.
+	 */
+	@Override
+	public BoundPojoModelPathPropertyNode<T, P> parent() {
```

Comment:
Well if you're doing that, it's all the more important to follow the getter convention for real getter, otherwise it's quite hard to understand what's going on.

I must admit that I have mixed feeling about things like `type()` as its purpose is not obvious (it's more obvious for `property(PropertyHandle)` for instance). But I can see how it's more readable.

Reply:
Ok; done for `parent` only then.
FYI the same naming conventions seem to be used in `java.nio.file.Path`: "get" prefix for actualy getters, but not for more complex methods.